torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1910,15 +1910,38 @@ def _backward_prologue(ctx, *flat_args):
                             "The grad inputs should be same number as forward output tangents"
                         )
 
-                    flat_processed_tangents = list(
+                    _num_mutated_inp_runtime = (
+                        CompiledFunction.metadata.num_mutated_inp_runtime_indices
+                    )
+
+                    flat_processed_tangents: List[torch.Tensor] = []
+                    if _num_mutated_inp_runtime > 0:
```

Comment:
hmm what is the new logic here for?

Reply:
Thanks, good catch. This is artifact from previous version, cleaning it.